pkg/reconciler/events/auditlogs/auditlogs.go

```diff
@@ -64,6 +70,28 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, s *v1alpha1.CloudAuditLo
 	s.Status.InitializeConditions()
 	s.Status.ObservedGeneration = s.Generation
 
+	// If GCP ServiceAccount is provided, configure workload identity.
+	if s.Spec.ServiceAccount != nil {
+		// Create corresponding k8s ServiceAccount if doesn't exist, and add ownerReference to it.
+		kServiceAccountName := psresources.GenerateServiceAccountName(s.Spec.ServiceAccount)
+		kServiceAccount, err := c.serviceAccountLister.ServiceAccounts(s.Namespace).Get(kServiceAccountName)
+		if err != nil {
+			if !apierrs.IsNotFound(err) {
```

Comment:
can we update the way we check for this so that is more consistent with other parts of the code?
In general (although not always) we do if IsNotFound(err) then we create it there, else if err != nil we return that err... else all good...
Examples:  https://github.com/knative/eventing/blob/d2b23626caef242492654bfdd3d8d1df2ae59af4/pkg/reconciler/channel/channel.go#L107
https://github.com/google/knative-gcp/blob/9b9b77969e2d503e6af1525bd9e0f2227b6395a1/pkg/reconciler/pubsub/pullsubscription/keda/pullsubscription.go#L122

Reply:
Should I create the service account here or just keep it nil. I didn't create the service account here is because I move the creation and add ownerReference in the pubsub base reconciler (I am not sure if keep it nil when it is not found also break the consistent...)